R/utils.R

```diff
@@ -150,3 +144,64 @@ update_field_names <- function(db_metadata, raw_or_label = 'raw'){
     unnest(cols = field_name_updated)
 }
 
+#' Update Multiple Choice Fields with Label Data
+#'
+#' @param db_data A REDCap database object
+#' @param db_metadata A REDCap metadata object
+#'
+#' @import dplyr
+#' @keywords internal
+
+multi_choice_to_labels <- function(db_data, db_metadata){
+
+  for (i in 1:nrow(db_metadata)) {
```

Comment:
I would put the following ahead of the `for` loop and then delete lines 162-184.

```r
# Handle columns where we change 0/1 to FALSE/TRUE (logical)
logical_cols <- db_metadata %>%
  filter(field_type %in% c("yesno", "truefalse", "checkbox")) %>%
  pull(field_name_updated)
 
db_data2 <- db_data %>%
  mutate(across(.cols = logical_cols, as.logical))
```

Reply:
Just want to make sure, you mean `db-data <- db_data` not `db_data2 <- db_data` right? Accidentally blindly copy/pasted and had test fails because we aren't using `db_data2` elsewhere haha